### ISSUE TITLE
Multiple workflows caused by instruction - Exercise 1, Task 4, Step 11

### ISSUE THREAD

**Issue by manasvin** [microsoft/MCW-Continuous-delivery-in-Azure-DevOps#120]
The instruction "Set up workflows for content-api and content-init in the same manner..." is causing multiple workflows to be created within Github Automation, as interpreted by the Labs team (who format/chop these labs for Immersion workshops). 

I believe it needs to be clarified although "fabrikam-api" and "fabrikam-init" yml files need to be created and patched/configured similar to "fabrikam-web", New worklfows are not needed. Simply creating new files does the job.

Attached image of the effect if the "Publish Docker Container" (step 5, Task 4, Ex 1) is selected again, as understood by the labs content editors.

![ex1-task4-problemresult](https://user-images.githubusercontent.com/18642248/159737072-d8d6ed9b-0018-4274-8961-e6ebf0ce1ae8.png)

**Comment by sadukie** [comment]
@manasvin  With regards to this particular issue, creating the files will still create the workflows. Whether you take the approach of creating a new workflow (repeating steps 5 through 10 for init and api) or create the files manually, you will still get multiple workflows appearing, as seen in your screenshot.

Is it confusing that the workflows are all being named Docker? If that's the problem, I can update the instructions to name the workflows.  That way, it could look something more like this:

![image](https://user-images.githubusercontent.com/106202/161816716-69410a3a-15f1-4932-b1d0-594604da5b6d.png)

**Comment by manasvin** [comment]
@sadukie I don't know where the confusion is. When I created newer files (by only adding a new file not new workflow) it definitely maintained a single workflow (called "Docker"). The problem with using the configure new workflows step is that all of them are trigger the build process individually so if there were three of them, the build process would be triggered **thrice** for the same update!

Multiple workflows definitely the incorrect step, especially on a hand on lab scenario meant to demonstrate good practice. Please fix.

**Comment by sadukie** [comment]
@manasvin When you wrote:

```
I believe it needs to be clarified although "fabrikam-api" and "fabrikam-init" yml files need to be created and patched/configured similar to "fabrikam-web", New worklfows are not needed. Simply creating new files does the job.
```

Where are you creating these new files that they aren't creating new workflows? If you put `fabrikam-api.yml`, `fabrikam-init.yml`, and `fabrikam-web.yml` in the `.github\workflows` folder, those will lead to 3 workflows.

Where are you creating these other files that are not creating additional workflows while still being executed?

**Comment by sadukie** [comment]
Ideally, in a production environment, the images would be in their own repos with their own workflows so that you only rebuild an image when the image is updated.

@manasvin  Are you asking to have the lab updated so that you are building all 3 images in one workflow? Something like this?
 
![One build - steps for extracting the metadata, building and pushing the image, signing the image, and post build for each environment (api, init, web)](https://user-images.githubusercontent.com/106202/162109910-2326df84-22c8-4ac3-bd3b-5ae4d245f4d9.png)

**Comment by sadukie** [comment]
If we wanted it to only update an image when there are changes to its respective folder in a single workflow, that workflow could look something like this:

![Check modified files and then trigger the respective build job](https://user-images.githubusercontent.com/106202/162113120-ea4874f2-39df-4be3-b6ab-96c6c14d0b3c.png)

However, going this route would require additional documentation and need to be approved by @DawnmarieDesJardins and @timahenning first.

**Comment by sadukie** [comment]
@manasvin Looking into this more, I will be making an update that takes the following approach:

![Checks for folder changes and updates the respective image](https://user-images.githubusercontent.com/106202/162114819-1ae45863-7f09-4117-ae56-a70fe31019c9.png)

In the yml file, there will be jobs for:

- check_changed_folders: Checks the `git diff` and sets output variables if a folder is changed - `run_api_job`, `run_web_job`, and `run_init_job`
- build-api: Runs only if `run_api_job` is set to true
- build-web: Runs only if `run_web_job` is set to true
- build-init: Runs only if `run_init_job` is set to true

Does this work for what you're trying to accomplish?

**Comment by manasvin** [comment]
@sadukie it may be easier If I could demonstrate these issues using the cloudlabs test instance (below) which is what we are
(https://github.com/CloudLabs-MCW/MCW-Continuous-delivery-in-Azure-DevOps/blob/prod/Hands-on%20lab/). 

is there a way to interactively connect via Teams/Zoom ?

for reference the page maintained by Cloudlabs (lab partner) with detailed instructions - https://github.com/CloudLabs-MCW/MCW-Continuous-delivery-in-Azure-DevOps/blob/prod/Hands-on%20lab/01.md to run the Immersion labs

in summary, when configure is used repeatedly multiple workflows are created. After any update to any of yml files trigger workflow actions on ALL the workflows (regardless of the name). Not ideal as a demonstrator. Multiple workflows are avoided if we  use the "configure" workflow only **once** and create yml files using the New file route.

**Comment by DawnmarieDesJardins** [comment]
Closing with merge of PR #125
